build-caching-maven-samples/enforcer-project/pom.xml

```diff
@@ -128,118 +137,31 @@
-                                <name>searchTransitive</name>
-                              </property>
-                              <property>
-                                <name>message</name>
-                              </property>
-                            </properties>
-                          </inputs>
-                        </property>
-                      </iteratedProperties>
-                      <nestedProperties>
-                        <property>
-                          <name>project</name>
-                          <iteratedProperties>
-                            <property>
-                              <name>dependencies</name>
-                              <inputs>
-                                <properties>
-                                  <property>
-                                    <name>groupId</name>
-                                  </property>
-                                  <property>
-                                    <name>artifactId</name>
-                                  </property>
-                                  <property>
-                                    <name>version</name>
-                                  </property>
-                                  <property>
-                                    <name>type</name>
-                                  </property>
-                                  <property>
-                                    <name>scope</name>
-                                  </property>
-                                </properties>
-                              </inputs>
-                            </property>
-                          </iteratedProperties>
-                        </property>
-                      </nestedProperties>
                       <outputs>
                         <cacheableBecause>banned dependency check should run when dependencies change</cacheableBecause>
                       </outputs>
                     </execution>
                     <execution>
                       <id>enforce-bytecode-version</id>
                       <inputs>
+                        <properties>
+                          <property>
+                            <name>rules</name>
+                          </property>
+                        </properties>
                         <ignoredProperties>
-                          <ignore>commandLineRules</ignore>
+                          <ignore>rulesToSkip</ignore>
+                          <ignore>rulesToExecute</ignore>
+                          <ignore>failIfNoRules</ignore>
                           <ignore>fail</ignore>
                           <ignore>failFast</ignore>
                           <ignore>ignoreCache</ignore>
                           <ignore>mojoExecution</ignore>
                           <ignore>session</ignore>
                           <ignore>skip</ignore>
+                          <ignore>project</ignore>
```

Comment:
Can you clarify your comment in the summary: 

> `project` was not ignored for the `enforce-bytecode-version` execution, but it was not being used in the execution configuration, so its input definition was removed and it was set as ignored

Just from the changeset, it seems `project` is being ignored where it wasn't before. It seems that you investigated a reason why this is ok, but I'm not quite following.